File: app/src/main/java/work/petrichor/petrichor/views/MainActivity.java
package work.petrichor.petrichor.views;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import java.io.IOException;

import work.petrichor.petrichor.R;
import work.petrichor.petrichor.modules.utils.AssetUtils;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        try {
            AssetUtils.copy(this);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void doPreview(View view) {
        startActivity(new Intent(this, PreviewActivity.class));
    }
}

File: app/src/main/java/work/petrichor/petrichor/views/MainActivity.kt
package work.petrichor.petrichor.views

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import work.petrichor.petrichor.R

import work.petrichor.petrichor.modules.utils.AssetUtils

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Thread(Runnable { if (!AssetUtils.copy(applicationContext))
            Toast.makeText(applicationContext, "无法拷贝模型，请检查权限！", Toast.LENGTH_LONG).show()
        }).start()
    }

    fun View.doStartPreview() = startActivity(Intent(this@MainActivity, PreviewActivity::class.java))
}
